File: Vault/System/Backend/Folder Structure Schema/Structure Builder Scripts/Notes Section/Inspo/builder_mj-inspo-description.js
#!/usr/bin/env node

// -----------------------------------------------------------------------------------
// 🎨 MidJourney Inspo Batch Folder Builder (Node.js Version)
// -----------------------------------------------------------------------------------
// USAGE:
//   node builder_mj-inspo-description.js --batch 3 --start 42 [--dry-run]
// -----------------------------------------------------------------------------------

const fs = require('fs');
const path = require('path');

// === Parse Arguments ===
const args = process.argv.slice(2);
let batch = 1;
let start = 1;
let dryRun = false;

args.forEach((arg, index) => {
    if (arg === '--batch') batch = parseInt(args[index + 1]);
    if (arg === '--start') start = parseInt(args[index + 1]);
    if (arg === '--dry-run') dryRun = true;
});

if (!batch || !start) {
    console.error('❌ Please provide both --batch and --start values.');
    process.exit(1);
}

console.log(`🔢 Creating ${batch} inspo folders starting at inspo_${String(start).padStart(4, '0')}`);
if (dryRun) {
    console.log('🧪 DRY-RUN mode enabled. No changes will be made.');
}

// === Helper Function ===
function createVarFolderAndFile(parent, varName) {
    const folderPath = path.join(parent, varName);
    const descFile = path.join(folderPath, `${varName}_description.txt`);

    if (dryRun) {
        console.log(`[DRY RUN] Would create folder: ${folderPath}`);
        console.log(`[DRY RUN] Would create file: ${descFile}`);
    } else {
        fs.mkdirSync(folderPath, { recursive: true });
        fs.writeFileSync(descFile, '');
        console.log(`📁 Created folder: ${folderPath}`);
        console.log(`📝 Created file: ${descFile}`);
    }
}

// === Main Loop ===
for (let i = 0; i < batch; i++) {
    const n = start + i;
    const inspoFolder = `inspo_${String(n).padStart(4, '0')}`;
    const fullPath = path.join('.', inspoFolder);

    console.log(`➡ Processing ${inspoFolder}`);

    if (fs.existsSync(fullPath)) {
        console.log(`⚠️ Folder already exists: ${fullPath}`);
    } else {
        if (dryRun) {
            console.log(`[DRY RUN] Would create base folder: ${fullPath}`);
        } else {
            fs.mkdirSync(fullPath, { recursive: true });
            console.log(`📂 Created base folder: ${fullPath}`);
        }
    }

    for (let v = 1; v <= 4; v++) {
        createVarFolderAndFile(fullPath, `var${v}`);
    }
}

if (dryRun) {
    console.log('✅ DRY-RUN complete. No files or folders created.');
} else {
    console.log(`✅ All ${batch} inspo folders created successfully starting at inspo_${String(start).padStart(4, '0')}`);
}

console.log('✨ All set! Happy creating.');

// -----------------------------------------------------------------------------------
// === End of Script ===
// -----------------------------------------------------------------------------------
// This script automates batch creation of MidJourney inspo folders.
// Each inspo folder contains 4 variant subfolders, each with a description file.
// Supports dry-run mode for safe previewing.
// -----------------------------------------------------------------------------------

File: Vault/System/Backend/Automations/Structure Builder Scripts/Notes Section/Inspo/builder_mj-inspo-description.rb
#!/usr/bin/env ruby

# -----------------------------------------------------------------------------------
# 🎨 MidJourney Inspo Batch Folder Builder - Ruby Version
# -----------------------------------------------------------------------------------
# USAGE:
#   ruby builder_mj-inspo-description.rb --batch 5 --start 7                  # Normal mode
#   ruby builder_mj-inspo-description.rb --batch 5 --start 7 --dry-run        # Preview only, no changes
#
# EXAMPLE:
#   ruby builder_mj-inspo-description.rb --batch 3 --start 42
# -----------------------------------------------------------------------------------

require 'fileutils'
require 'optparse'

# -----------------------------------------------------------------------------------
# === Argument Parsing ===
# -----------------------------------------------------------------------------------

options = {}
OptionParser.new do |opts|
  opts.banner = "🎨 MidJourney Inspo Batch Folder Builder - Ruby Version"

  opts.on("--batch BATCH", Integer, "Number of inspo folders to create") do |v|
    options[:batch] = v
  end

  opts.on("--start START", Integer, "Starting number for folder naming") do |v|
    options[:start] = v
  end

  opts.on("--dry-run", "Preview actions without making changes") do
    options[:dry_run] = true
  end
end.parse!

batch = options[:batch] || (abort "❌ You must specify --batch")
start = options[:start] || (abort "❌ You must specify --start")
dry_run = options[:dry_run] || false

BASE_PATH = "./"

puts "🔢 Creating #{batch} inspo folders starting at inspo_%04d" % start

puts "🧪 DRY-RUN mode enabled. No changes will be made." if dry_run

# -----------------------------------------------------------------------------------
# === Helper Method ===
# -----------------------------------------------------------------------------------

def create_var_folder_and_file(parent, var_name, dry_run)
  folder_path = File.join(parent, var_name)
  desc_file = File.join(folder_path, "#{var_name}_description.txt")

  if dry_run
    puts "[DRY RUN] Would create folder: #{folder_path}"
    puts "[DRY RUN] Would create file: #{desc_file}"
  else
    FileUtils.mkdir_p(folder_path)
    FileUtils.touch(desc_file)
    puts "📁 Created folder: #{folder_path}"
    puts "📝 Created file: #{desc_file}"
  end
end

# -----------------------------------------------------------------------------------
# === Main Loop ===
# -----------------------------------------------------------------------------------

batch.times do |i|
  n = start + i
  inspo_folder = "inspo_%04d" % n
  full_path = File.join(BASE_PATH, inspo_folder)

  puts "➡ Processing #{inspo_folder}"

  if Dir.exist?(full_path)
    puts "⚠️ Folder already exists: #{full_path}"
  else
    if dry_run
      puts "[DRY RUN] Would create base folder: #{full_path}"
    else
      FileUtils.mkdir_p(full_path)
      puts "📂 Created base folder: #{full_path}"
    end
  end

  (1..4).each do |v|
    create_var_folder_and_file(full_path, "var#{v}", dry_run)
  end
end

# -----------------------------------------------------------------------------------
# === Completion Message ===
# -----------------------------------------------------------------------------------
if dry_run
  puts "✅ DRY-RUN complete. No files or folders created."
else
  puts "✅ All #{batch} inspo folders created successfully starting at inspo_%04d" % start
end

puts "✨ All set! Happy creating."

# -----------------------------------------------------------------------------------
# === End of Script ===
# -----------------------------------------------------------------------------------
# This script automates batch creation of MidJourney inspo folders.
# Each inspo folder contains 4 variant subfolders, each with a description file.
# Supports dry-run mode for safe previewing.
#
# Educational and creative use only. Use at your own risk.
# -----------------------------------------------------------------------------------
